### PR TITLE
SPP v 5 and Login timeout  

### PR DESCRIPTION
**Pull request by MrGenga**
### Description

Update to new SPP protocol.

### Reason to modify

Because, Nemisys is working on the last SPP protocol.

### Tests & Reviews

I have tested code, and it works. Test server: pe.gc-game.ru port: 19132. Lobby and RobBank mini-game is on Genisys, BedWars on Nukkit with SynapseAPI.

### CODE REVIEW

**Comment by gaoyichuan** [comment]
What's the commit 'Start a Pull Request' for?
Code Review -1

**Comment by ishitatsuyuki** [comment]
Poor formatting, do not add meaningless commit.

**Comment by MrGenga** [comment]
Sorry, but I am not master in git program.

**Comment by MrGenga** [comment]
And formating is normal -_-

File: src/synapse/network/synlib/ServerConnection.php [hunk 1]
```diff
@@ -70,12 +67,16 @@ private function tickProcessor(){
...
+		if(($data = $this->readPacket()) !== null){
+			foreach($data as $pk){
+				$this->server->pushThreadToMainPacket($pk);
+			}
+                        $data = null; //Hm.... :/
 		}
 		while(strlen($data = $this->server->readMainToThreadPacket()) > 0){
 			$this->writePacket($data);
 		}
+                $data = null; //Hmmmmm....... :|
```
Comment (dktapps) [inline src/synapse/network/synlib/ServerConnection.php]: Formatting, again

File: src/synapse/network/synlib/ServerConnection.php [hunk 2]
```diff
@@ -133,33 +130,37 @@ public function getSocket(){
...
-			$buffer = substr($buffer, 4);//截取剩余的缓存，长度后面的都是包的内容了
-			if($len != strlen($buffer)){//校验长度
-				throw new \Exception("Wrong packet $buffer");
 			}
-			return $buffer;
+			if($offset < $len){
+                $this->receiveBuffer = substr($this->receiveBuffer, $offset);
+            }else{
+                $this->receiveBuffer = "";
+             }
```
Comment (dktapps) [inline src/synapse/network/synlib/ServerConnection.php]: and here

**Comment by dktapps** [comment]
You don't seem to have changed a thing from your last 2 PRs (minus the fact there are now less commits, which is the only good thing I have to say since). There are still unprofessional comments littered around and the formatting is still a mess. Code Review -1.

**Pull request closed without merging**